core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java

```diff
@@ -702,7 +702,7 @@ public Result visit(TableModify modify) {
 
       final SqlInsert sqlInsert =
           new SqlInsert(POS, SqlNodeList.EMPTY, sqlTargetTable, sqlSource,
-              identifierList(modify.getInput().getRowType().getFieldNames()));
+              identifierList(modify.getTable().getRowType().getFieldNames()));
 
```

Comment:
Shouldn't we use `TableModify.getUpdateColumnList` ? Also add a test for insert with explicit table columns.

Reply:
Thanks, Danny~
I will refine by your comments soon.

Reply:
`TableModify.getUpdateColumnList` returns null for `Operation.INSERT` [1]
I added a test for insert with explicit table columns.

[1] https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/rel/core/TableModify.java#L125